Azure/azure-cli
# [App Service] Fix#22934: `az functionapp delete`: Add new parameter `--keep-empty-plan` to support keeping empty app service plan

**Related command**
az functionapp delete

**Description**
In the `az functionapp delete` command, this PR adds support for the `--keep-empty-plan` argument that would allow customers to keep the App Service plan even after deleting the last function app in the plan. This is to resolve issue #22934

**Testing Guide**

1. Create an App Service plan (`az functionapp plan create -g <resource-group-name> -n <plan-name> --sku S1`)
2. Create a function app (`az functionapp create -g <resource-group-name> -n <function-app-name> --plan <plan-name> -s <storage-account-name>`)
3. Delete function app (`az functionapp delete -g <resource-group-name> -n <function-app-name> --keep-empty-plan`)
4. Check the Azure Portal and you should see the App Service plan remains even though the function app has been deleted

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.
